packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_router.ts

```diff
@@ -6,15 +6,15 @@
  * Side Public License, v 1.
  */
 
-import type { IRouter } from '@kbn/core-http-server';
 import type { VersionedRouter, VersionedRoute, VersionedRouteConfig } from '@kbn/core-http-server';
 import { omit } from 'lodash';
 import { CoreVersionedRoute } from './core_versioned_route';
 import type { HandlerResolutionStrategy, Method, VersionedRouterRoute } from './types';
+import type { Router } from '../router';
 
 /** @internal */
 export interface VersionedRouterArgs {
-  router: IRouter;
+  router: Router;
```

Comment:
(thinking out loud) I get why we did that, I still can't avoid thinking it's a shame that we're now forced to use our concrete implementation directly here. Relation was pure before, it's no longer.

Reply:
I agree, it is a :shame: moment. One alternative is that we can something like an `InternalRegistrar` interface to keep things more pure. Wdyt?